compaheuiler/src/c_bigint_runtime.c

```
@@ -0,0 +1,125 @@
+CBIG_INLINE int64_t tag_mul(int64_t a, int64_t b) {
+    if ((a & b & 1) != 0) { int64_t r; if (try_mul(a >> 1, b >> 1, &r) && r >= SMALL_MIN && r <= SMALL_MAX) return tag_small(r); }
+    return cbig_mul(a, b);
+}
+
+typedef struct {
+    int64_t queue[QUEUE_CAP]; size_t q_head, q_len;
+    int64_t port[PORT_CAP]; size_t p_len; int64_t port_last;
+    int big_mode;
+} SpecialStorage;
+static void sp_init(SpecialStorage *s) { s->q_head=0; s->q_len=0; s->p_len=0; s->port_last=0; s->big_mode=0; }
+static void sp_push(SpecialStorage *s, size_t sel, int64_t v) { if (sel==21) { s->queue[(s->q_head+s->q_len)%QUEUE_CAP]=v; s->q_len++; } else { s->port_last=v; s->port[s->p_len++]=v; } }
```

Cppcheck (2.21.0)

[style] The function 'sp_push' is never used.

(unusedFunction)